lib/shopify_app/controller_concerns/embedded_app.rb

```diff
@@ -0,0 +1,19 @@
+module ShopifyApp
+  module EmbeddedApp
+    extend ActiveSupport::Concern
+
+    included do
+      if ShopifyApp.configuration.embedded_app?
+        after_action :set_esdk_headers
+        layout 'embedded_app'
+      end
+    end
+
+    private
+
+    def set_esdk_headers
+      response.set_header('P3P', 'CP="Not used"')
+      response.default_headers.delete('X-Frame-Options')
```

Comment:
For what its worth I did try without the headers and the ESDK does not work so this is still required